packages/template-retail-react-app/app/pages/order-status/index.test.js

```diff
@@ -5,13 +5,146 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import React from 'react'
-import {render, screen} from '@testing-library/react'
+import {screen} from '@testing-library/react'
 import OrderStatusPage from '@salesforce/retail-react-app/app/pages/order-status/index.jsx'
+import userEvent from '@testing-library/user-event'
+import {renderWithProviders} from '@salesforce/retail-react-app/app/utils/test-utils'
+
+// Mock the navigation hook
+const mockNavigate = jest.fn()
+jest.mock('@salesforce/retail-react-app/app/hooks/use-navigation', () => ({
+    __esModule: true,
+    default: () => mockNavigate
+}))
+
+// Mock the useCurrentCustomer hook
+const mockUseCurrentCustomer = jest.fn()
+jest.mock('@salesforce/retail-react-app/app/hooks/use-current-customer', () => ({
+    __esModule: true,
+    useCurrentCustomer: () => mockUseCurrentCustomer()
+}))
 
 describe('OrderStatusPage', () => {
-    it('renders the order status page', () => {
-        render(<OrderStatusPage />)
-        expect(screen.getByTestId('order-status-page')).toBeTruthy()
+    const user = userEvent.setup()
+
+    beforeAll(() => {
+        jest.spyOn(console, 'error').mockImplementation(() => {})
+    })
+
+    afterAll(() => {
+        console.error.mockRestore()
+    })
+
+    beforeEach(() => {
+        jest.clearAllMocks()
+        // Default mock for guest user
+        mockUseCurrentCustomer.mockReturnValue({
+            data: {
+                isRegistered: false,
+                customerType: 'guest'
+            }
+        })
+    })
+
+    test('renders the order status page with correct structure', () => {
```

Comment:
```suggestion
    test('displays order status page to users with main heading', () => {
```

Focuses moreon what user experiences (seeing page and heading) rather than how the code works